coverage.py

```diff
@@ -450,192 +576,51 @@ def ensure_coverage(self, item, force=False):
         else:
             coverage_record = None
         return coverage_record
-
-    @property
-    def output_source(self):
-        """Look up the DataSource object corresponding to the
-        service we're running this data through.
-
-        Out of an excess of caution, we look up the DataSource every
-        time, rather than storing it, in case a CoverageProvider is
-        ever used in an environment where the database session is
-        scoped (e.g. the circulation manager).
-        """
-        return DataSource.lookup(self._db, self.output_source_name)
-
-    def license_pool(self, identifier):
-        """Finds or creates the LicensePool for a given Identifier."""
-        license_pool = identifier.licensed_through
-        if not license_pool:
-            if self.CAN_CREATE_LICENSE_POOLS:
-                # The source of this data also provides license
-                # pools, so it's okay to automatically create
-                # a license pool for this book.
-                license_pool, ignore = LicensePool.for_foreign_id(
-                    self._db, self.output_source, identifier.type, 
-                    identifier.identifier
-                )
-            else:
-                return None
-        return license_pool
-
+    
     def edition(self, identifier):
-        """Finds or creates the Edition for a given Identifier."""
-        license_pool = self.license_pool(identifier)
-        if not license_pool:
-            e = "No license pool available"
-            return CoverageFailure(identifier, e, data_source=self.output_source, transient=True)
-
+        """Finds or creates an Edition representing this coverage provider's
+        view of a given Identifier.
+        """
         edition, ignore = Edition.for_foreign_id(
-            self._db, license_pool.data_source, identifier.type,
+            self._db, self.data_source, identifier.type,
             identifier.identifier
         )
         return edition
 
-    def work(self, identifier):
-        """Finds or creates the Work for a given Identifier.
-        
-        :return: The Work (if it could be found) or an appropriate
-        CoverageFailure (if not).
-        """
-        license_pool = self.license_pool(identifier)
-        if not license_pool:
-            e = "No license pool available"
-            return CoverageFailure(identifier, e, data_source=self.output_source, transient=True)
-        work, created = license_pool.calculate_work(even_if_no_author=True)
-        if not work:
-            e = "Work could not be calculated"
-            return CoverageFailure(identifier, e, data_source=self.output_source, transient=True)
-        return work
-
-
-    def set_metadata(self, identifier, metadata, 
```

Comment:
This was _not_ moved to CollectionCoverageProvider, it was just moved below this part of the diff.